gui/src/2D/controls/selector.ts

```diff
@@ -0,0 +1,528 @@
+        }); 
+    
+        var _selector = Control.AddHeader(button, text, "200px", { isHorizontal: true, controlFirst: true });
+        _selector.height = "30px";
+        _selector.horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+        _selector.left = "4px";
+    
+        this._groupPanel.addControl(_selector);
+        this._selectors.push(_selector);    
+        button.isChecked = checked;
+
+        if(this._groupPanel.parent) {
+            if(this._groupPanel.parent.parent) {
+                button.color = this._groupPanel.parent.parent.buttonColor;
+                button.background  = this._groupPanel.parent.parent.buttonBackground;
+            }
+        } 
+    }
+
+    /** Adds a radio button as a control
+     * @param text is the label for the selector
+     * @param func is the function called when the Selector is checked
+     * @param checked is true when Selector is checked
+     */
+    protected _addRadio(text: string, nb: number, name: string, func: (n: number)=>any, checked: boolean) {				
+        checked = checked || false;
+        var button = new RadioButton();
+        button.name = text;
+        button.width = "20px";
+        button.height = "20px";
+        button.color = "#364249";
+        button.background = "#CCCCCC"; 
+        button.group = name;
+        button.horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+
+        button.onIsCheckedChangedObservable.add(function(state) {                       		                  
+            if(state) {
+                func(nb);
+            }
+        });
+        
+        var _selector = Control.AddHeader(button, text, "200px", { isHorizontal: true, controlFirst: true });
+        _selector.height = "30px";
+        _selector.horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+        _selector.left = "4px";
+        this._groupPanel.addControl(_selector);
+        this._selectors.push(_selector);
+        button.isChecked = checked;
+        
+        if(this._groupPanel.parent) {
+            if(this._groupPanel.parent.parent) {
+                button.color = this._groupPanel.parent.parent.buttonColor;
+                button.background  = this._groupPanel.parent.parent.buttonBackground;
+            }
+        } 
+    }
+
+    /**
+     * Adds a slider as a control
+     * @param text is the label for the SliderBar
+     * @param func is the function called when the Slider moves
+     * @param unit is a string describing the units used, eg degrees or metres
+     * @param min is the minimum value for the Slider
+     * @param max is the maximum value for the Slider
+     * @param value is the start value for the Slider between min and max
+     * @param onVal is the function used to format the value displayed, eg radians to degrees
+     */
+    protected _addSldr(text: string, func: (v: any)=>any, unit: string, min: number, max:number, value: number, onValueChange: (v: number)=>number) {
+        var button = new Slider();  

```

Comment:
I may have missed something but the control seems not added to the sliders array

Reply:
Now added to selectors array